### PR TITLE
Secure enclave signer

### PR DESCRIPTION
**Pull request by johanfylling**
Adding possibility so create JWS:s signed with keys in the Secure Enclave.

### CODE REVIEW

**Review by appfrilans — APPROVED** [review]

File: JOSESwift/Sources/SecureEnclaveSigner.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
```
Comment (travisspencer) [inline JOSESwift/Sources/SecureEnclaveSigner.swift]: This should probably be the Apache 2 license

File: JOSESwift/Sources/Signer.swift [hunk 2]
```diff
@@ -53,11 +54,14 @@ struct Signer<KeyType> {
             // swiftlint:disable:next force_cast
             self.signer = RSASigner(algorithm: signingAlgorithm, privateKey: privateKey as! RSASigner.KeyType)
         case .ES256, .ES384, .ES512:
-            guard type(of: privateKey) is ECSigner.KeyType.Type else {
+            if type(of: privateKey) is ECSigner.KeyType.Type {
+                self.signer = ECSigner(algorithm: signingAlgorithm, privateKey: privateKey as! ECSigner.KeyType)
+            } else if #available(iOS 13.0, *),
+                      type(of: privateKey) is SecureEnclave.P256.Signing.PrivateKey.Type {
+                self.signer = SecureEnclaveSigner(algorithm: signingAlgorithm, privateKey: privateKey as! SecureEnclave.P256.Signing.PrivateKey)
```
Comment (travisspencer) [inline JOSESwift/Sources/Signer.swift]: long line

File: JOSESwift/Sources/Verifier.swift [hunk 3]
```diff
@@ -55,10 +56,14 @@ struct Verifier {
             // swiftlint:disable:next force_cast
             self.verifier = RSAVerifier(algorithm: verifyingAlgorithm, publicKey: publicKey as! RSAVerifier.KeyType)
         case .ES256, .ES384, .ES512:
-            guard type(of: publicKey) is ECVerifier.KeyType.Type else {
+            if type(of: publicKey) is ECVerifier.KeyType.Type {
+                self.verifier = ECVerifier(algorithm: verifyingAlgorithm, publicKey: publicKey as! ECVerifier.KeyType)
+            } else if #available(iOS 13.0, *),
+                      type(of: publicKey) is P256.Signing.PublicKey.Type {
+                self.verifier = SecureEnclaveVerifier(algorithm: verifyingAlgorithm, publicKey: publicKey as! P256.Signing.PublicKey)
```
Comment (travisspencer) [inline JOSESwift/Sources/Verifier.swift]: here too

**Review by travisspencer — APPROVED** [review]

File: JOSESwift/Sources/SecureEnclaveSigner.swift (hunk 1 above)
Comment (johanfylling) [reply JOSESwift/Sources/SecureEnclaveSigner.swift]: Yea, I was unsure about what to put here.

Comment (travisspencer) [reply JOSESwift/Sources/SecureEnclaveSigner.swift]: You can find the Apache one in one of the other open source plug-ins/projects.

**Pull request merged**